packages/paste-core/components/chat-log/src/useChatLogger.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import {uid} from '@twilio-paste/uid-library';
+
+import type {MessageVariants} from './MessageVariantContext';
+
+type PushChat = (chat: PartialIDChat) => void;
+type PopChat = (id?: string) => void;
+
+export type Chat = {
+  id: string;
+  variant?: MessageVariants;
+  content: React.ReactElement;
+};
+
+export type PartialIDChat = Omit<Chat, 'id'> & Partial<Pick<Chat, 'id'>>;
```

Comment:
Question: why do you have this partialIDChat type? could you just mark id as not required in the type above?

Reply:
Ohhhh.... I see. Thanks for explaining!